lightkube/core/generic_client.py

```diff
@@ -192,6 +193,14 @@ def raise_for_status(resp):
     def build_adapter_request(self, br: BasicRequest):
         return self._client.build_request(br.method, br.url, params=br.params, json=br.data, headers=br.headers)
 
+    def convert_to_resource(self, res: Type[r.Resource], item: dict) -> r.Resource:
+        resource_def = res._api_info.resource
```

Comment:
I didn't use `r.api_info` because it appears to expect an `r.Resource` instance (vs. a `Type[r.Resource]`).  

I _could_ change `r.api_info`to expect `Union[r.Resource, Type[r.Resource]]` to avoid accessing an internal field here, let me know.